distro/run/assembly/resources/run.bat

```diff
@@ -34,14 +34,18 @@ IF "x%JAVA_HOME%" == "x" (
       GOTO :EOF
     )
     ECHO Setting JAVA property to "%JAVA_HOME%\bin\java"
-    SET JAVA="%JAVA_HOME%\bin\java"
+    SET "JAVA=%JAVA_HOME%\bin\java"
+    SET "TEMP_PATH=%PATH%"
+    REM add temp the JAVA_HOME so this java is used in the java version check
+    SET "PATH=%JAVA_HOME%\bin;%PATH%"
   )
 )
 
 SET EXPECTED_JAVA_VERSION=17
-FOR /f "tokens=3" %%g IN ('JAVA -version 2^>^&1 ^| findstr /i "version"') DO (
+FOR /f "tokens=3" %%g IN ('java -version 2^>^&1 ^| findstr /i "version"') DO (
```

Comment:
```suggestion
FOR /f "tokens=3" %%g IN ('%JAVA_HOME%\bin\java -version 2^>^&1 ^| findstr /i "version"') DO (
```
❓ Would it be possible to do something like this ⬆️ instead of changing the PATH variable?

Reply:
I tried several options to consider the JAVA_HOME on this row. However, I was not able to cover the use case with empty space on the path, that failed in the for loop, no matter if I used JAVA or JAVA_HOME and combination of quotes.